### DISCUSSION TITLE
CheckboxValues property does not update until page is changed

### DISCUSSION THREAD

**Discussion by corey-Robinson1337** [Power-Components/livewire-powergrid#1285] [Q&A]
I'm trying to react to the count of the number of items selected in real time, but this isn't possible because the checkboxValues property isn't reactive. Any suggestions for this?

The value of checkboxValues isn't updated when a checkbox is changed, only when you interact with pagination for some reason.

**Comment by abhaypithadiya** [comment]
Hey!

You can use ```window.pgBulkActions.get(your_table_name)``` to retrieve selected rows in Javascript in real time, this won't send a request to the server.

If you have named your table you can change the table name or it will be 'default'.

Hope this helps!

**Reply by corey-Robinson1337** [reply]
I see do you have any examples of how to listen for the the checkbox clicks and then send that back to the server by dispatching an event?

**Reply by abhaypithadiya** [reply]
Hey! You can achieve this with just livewire as well, but i feel this is a wrong way to do it but if this helps you out, you can try the following:

1. Publish the views with the command ```php artisan vendor:publish --tag=livewire-powergrid-views```

2. Go to file ```resources/views/vendor/livewire-powergrid/components/checkbox-row.blade.php```

3.  On line 46, add '.live' to wire.model, ```wire:model.live="checkboxValues"```

4. In your controller you can then use lifecycle hook to get the ids, something like this
```
    public function updatedCheckboxValues($value)
    {
        dd($value);
    }

```

This will call the server every time someone checks or unchecks the checkbox, you can play around this code and find a solution. 

I hope this helps you in someway.
